jinja2/filters.py

```diff
@@ -940,6 +940,37 @@ def do_attr(environment, obj, name):
     return environment.undefined(obj=obj, name=name)
 
 
+@contextfilter
+def do_applymacro(*args, **kwargs):
+    """Applies a macro by reference instead of directly.
+
+    Calling a macro directly like,
+
+    {{ event_detail(event) }}
+
+    Is the equivalent with applymacro,
+
+    {{ event|applymacro(macroname='event_detail') }}
+
+    or
+
+    {{ event|applymacro(macro=event_detail) }}
+
+    Which then allows you to use the macro in other things like map,
+
+    {{ events|map('applymacro', macro=event_detail)|join(', ') }}
+
+    """
+    context = args[0]
+    args = args[1:]
+
+    macro = kwargs.pop('macro', None)
+    if macro is None:
+        macro = context.vars[kwargs.pop('macroname')]
+
+    return macro(*args, **kwargs)
```

Comment:
Is this the correct way to call macros? This probably requires `context.call()` or an equivalent for macros.

Reply:
I'm not sure.  It *works*...  I'd need a pointer to the correct way, if there is one.

Reply:
It looks like other filters don't use use it even if they take context, so we can skip that for now.